src/main/java/org/apache/log4j/ConsoleAppender.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * Modifications by Red Hat, Inc.
+ *
+ * This file incorporates work covered by the following notice(s):
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.log4j;
+
+import java.io.FileDescriptor;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintStream;
+
+import org.apache.log4j.helpers.LogLog;
+
+/**
+ * ConsoleAppender appends log events to <code>System.out</code> or
+ * <code>System.err</code> using a layout specified by the user. The
+ * default target is <code>System.out</code>.
+ *
+ * <p>
+ * <p>
+ *    Modification was done in JBoss fork to use the real <strong>stdout</strong>/<strong>stderr</strong>
+ *    via <code>FileDescriptor.out</code>/<code>FileDescriptor.err</code> instead of
+ *    <code>System.out</code>/<code>System.err</code>, because in some cases, the standard
+ *    stream may be reassigned.
+ * <p>
+ * @author Ceki G&uuml;lc&uuml;
+ * @author Curt Arnold
+ * @author <a href="mailto:lgao@redhat.com">Lin Gao</a>
+ * @since 1.1 */
+public class ConsoleAppender extends WriterAppender {
+
+  public static final String SYSTEM_OUT = "System.out";
+  public static final String SYSTEM_ERR = "System.err";
+
+  private static final PrintStream out = new PrintStream(new FileOutputStream(FileDescriptor.out), true);
+  private static final PrintStream err = new PrintStream(new FileOutputStream(FileDescriptor.err), true);
```

Comment:
These should probably be `new PrintStream(new BufferedOuputStream(new FileOutputStream(FileDescriptor.xxx), 128));` to keep in sync what the JDK does.